/builds/fossdd/aports/community/kodi/src/xbmc-21.2-Omega/xbmc/utils/EndianSwap.h:31:64: error: unsupported inline asm: input with type 'int' matching output with type 'uint16_t' (aka 'unsigned short')
   31 |         __asm__("rlwimi %0,%2,8,16,23" : "=&r" (result) : "0" (x >> 8), "r" (x));
      |                                                 ~~~~~~         ^~~~~~
---
--- a/xbmc/utils/EndianSwap.h
+++ b/xbmc/utils/EndianSwap.h
@@ -23,25 +23,6 @@
 extern "C" {
 #endif
 
-#if defined(__GNUC__) && (defined(__powerpc__) || defined(__ppc__))
-static __inline__ uint16_t Endian_Swap16(uint16_t x)
-{
-        uint16_t result;
-
-        __asm__("rlwimi %0,%2,8,16,23" : "=&r" (result) : "0" (x >> 8), "r" (x));
-        return result;
-}
-
-static __inline__ uint32_t Endian_Swap32(uint32_t x)
-{
-        uint32_t result;
-
-        __asm__("rlwimi %0,%2,24,16,23" : "=&r" (result) : "0" (x>>24), "r" (x));
-        __asm__("rlwimi %0,%2,8,8,15"   : "=&r" (result) : "0" (result),    "r" (x));
-        __asm__("rlwimi %0,%2,24,0,7"   : "=&r" (result) : "0" (result),    "r" (x));
-        return result;
-}
-#else
 static __inline__ uint16_t Endian_Swap16(uint16_t x) {
         return((x<<8)|(x>>8));
 }
@@ -49,7 +30,6 @@ static __inline__ uint16_t Endian_Swap16(uint16_t x) {
 static __inline__ uint32_t Endian_Swap32(uint32_t x) {
         return((x<<24)|((x<<8)&0x00FF0000)|((x>>8)&0x0000FF00)|(x>>24));
 }
-#endif
 
 static __inline__ uint64_t Endian_Swap64(uint64_t x) {
         uint32_t hi, lo;

